test/find_test.dart

```diff
@@ -0,0 +1,117 @@
+              expect(mod.index.value.toInt(), 0);
+            }),
+            test('at random position', () {
+              final bus = Const(bin('11111100'), width: 8);
+              final mod = Find(bus);
+              expect(mod.index.value.toInt(), 2);
+            }),
+            test('at last position', () {
+              final bus = Const(bin('10000000'), width: 8);
+              final mod = Find(bus);
+              expect(mod.index.value.toInt(), 7);
+            }),
+            test('when first one is not present', () {
+              final bus = Const(bin('00000000'), width: 8);
+              final mod = Find(bus);
+              // When your find is not found it will result in error
+              expect(mod.error.value.toInt(), 1);
+            }),
+          });
+
+  group(
+      'find first zero with n as null',
+      () => {
+            test('at 0th position', () {
+              final bus = Const(bin('11010100'), width: 8);
+              final mod = Find(bus, countOne: false);
+              expect(mod.index.value.toInt(), 0);
+            }),
+            test('at random position', () {
+              final bus = Const(bin('10101011'), width: 8);
+              final mod = Find(bus, countOne: false);
+              expect(mod.index.value.toInt(), 2);
+            }),
+            test('at last position', () {
+              final bus = Const(bin('01111111'), width: 8);
+              final mod = Find(bus, countOne: false);
+              expect(mod.index.value.toInt(), 7);
+            }),
+            test('when first zero is not present', () {
+              final bus = Const(bin('11111111'), width: 8);
+              final mod = Find(bus, countOne: false);
+              // When your find is not found it will result in error
+              expect(mod.error.value.toInt(), 1);
+            }),
+          });
+
+  group(
+      'find nth zero',
+      () => {
+            test('n is 0', () {
+              final bus = Const(bin('11010100'), width: 8);
+              final mod = Find(bus, countOne: false);
+              expect(mod.index.value.toInt(), 0);
+            }),
+            test('when n is 2 (find 3rd zero; n is zero index)', () {
+              final bus = Const(bin('10101011'), width: 8);
+              final mod = Find(bus,
+                  countOne: false, n: Const(2, width: log2Ceil(8) + 1));
+              expect(mod.index.value.toInt(), 6);
+            }),
+            test('n is outside bound', () {
+              final bus = Const(bin('00000000'), width: 8);
+              final mod =
+                  Find(bus, countOne: false, n: Const(10, width: log2Ceil(10)));
+              expect(mod.error.value.toInt(), 1);
+            }),
+            test('if all 0s', () {
+              final bus = Const(bin('00000000'), width: 8);
+              final mod =
+                  Find(bus, countOne: false, n: Const(7, width: log2Ceil(8)));
+              expect(mod.index.value.toInt(), 0);
```

Comment:
shouldn't the 7th `0` be in position `7` here?

Reply:
seem n is of width `log2Ceil(8) + 1` was also missing

Reply:
done